File: python/hackerrank/simulation/serviceLane.py

def serviceLane():
    n, t = map(int, input().split())
    width = list(map(int, input().split()))

    for _ in range(t):
        start, end = map(int, input().split())
        print(min(width[i] for i in range(start, end + 1)))

if __name__ == "__main__":
    serviceLane()

File: golang/hackerrank/simulation/serviceLane.go
package simulation

func ServiceLane(n int, cases [][]int, width []int) []int {
	var result []int

	for _, c := range cases {
		start := c[0]
		end := c[1]

		minWidth := width[start]
		for i := start + 1; i <= end; i++ {
			if width[i] < minWidth {
				minWidth = width[i]
			}
		}
		result = append(result, minWidth)
	}

	return result
}
